mongodump/mongodump_test.go

```diff
@@ -1188,23 +1239,44 @@ func TestMongoDumpCollectionOutputPath(t *testing.T) {
 			So(len(fileComponents), ShouldEqual, 3)
 
 			filePath := fileComponents[len(fileComponents)-1]
-			So(len(filePath), ShouldEqual, len(colName))
-			So(filePath, ShouldNotContainSubstring, util.EscapeCollectionName("$"))
+			So(filePath, ShouldEqual, colName)
+			So(filePath, ShouldNotContainSubstring, "%24")
 		})
 
-		Convey("truncate a collection name that results in an output path > 255 bytes", func() {
+		Convey("truncate a collection name that results in an output path >255 bytes", func() {
 			md.OutputOptions.Out = "dump"
 
-			// 26 bytes * 10 = 260 bytes > 255 bytes
-			// (output path is already over the limit, regardless of file extension)
-			colName := strings.Repeat("abcdefghijklmnopqrstuvwxyz", 10)
+			Convey("without special characters", func() {
+				// 26 bytes * 10 = 260 bytes > 238 bytes
+				// (output path is already over the file name limit of 255, regardless of file extension)
+				colName := strings.Repeat("abcdefghijklmnopqrstuvwxyz", 10)
 
-			fileComponents := strings.Split(md.outputPath(testDB, colName), "/")
-			So(len(fileComponents), ShouldEqual, 3)
+				fileComponents := strings.Split(md.outputPath(testDB, colName), "/")
+				So(len(fileComponents), ShouldEqual, 3)
 
-			filePath := fileComponents[len(fileComponents)-1]
-			So(len(filePath), ShouldEqual, 238)
-			So(filePath, ShouldContainSubstring, util.EscapeCollectionName("$"))
+				filePath := fileComponents[len(fileComponents)-1]
+				So(filePath[:211], ShouldEqual, colName[:208]+"%24")
+
+				hashDecoded, _ := base64.RawURLEncoding.DecodeString(filePath[211:])
+				hash := sha1.Sum([]byte(colName))
+				So(bytes.Compare(hashDecoded, hash[:]), ShouldEqual, 0)
+			})
+
+			Convey("with special characters", func() {
+				// (26 bytes + 3 special bytes) * 8 = 232 bytes < 238 bytes
+				// (output path is under the limit, but will go over when we escape the special symbols)
+				colName := strings.Repeat("abcdefghijklmnopqrstuvwxyz+/@", 8)
```

Comment:
I'm grouping the unit tests for a '/' symbol in the collection name and encoding to a length >238 bytes. That seems okay to me, but I can separate them if you guys think that's better.